lib/ansible/modules/network/nxos/nxos_vrf.py

```diff
@@ -344,12 +344,13 @@ def map_obj_to_commands(updates, module):
 
 
 def validate_vrf(name, module):
-    if name == 'default':
-        module.fail_json(msg='cannot use default as name of a VRF')
-    elif len(name) > 32:
-        module.fail_json(msg='VRF name exceeded max length of 32', name=name)
-    else:
-        return name
+    if name:
+        if name == 'default':
```

Comment:
can name attribute has value 'default  ' ?  if so this will get pass through.

Reply:
Good catch.

Reply:
@rohitthakur2590 , even if we give name = "default " (with space) , the device accepts it and removes space. The PR actually fixes the issue faced, while aggregate field is empty.

Reply:
@GomathiselviS yeah it looks correct for  "aggregate field is empty" scenario, but then checking for 'default' becomes less relevant if it would let it pass with couple of spaces.